fix(linter): set package version in `package.json`

overlookmotel opened the pull request:
The version field is used by NAPI build. Bring it up to date.

#13895 updates the pre-release workflow so it's kept in sync in future.

camc314 commented:
will this be confusing that it's different to the oxlint version (1.17.x for example)

camc314 reviewed (approved):
does this get bumped automagically when we do oxlint release

overlookmotel committed:
fix(linter): set package version in `package.json` (#13890)

The version field is used by NAPI build. Bring it up to date.

#13895 updates the pre-release workflow so it's kept in sync in future.
